lib/marc_to_argot/macros/shared/included_work.rb

```diff
@@ -0,0 +1,304 @@
+              included_work['label'] = label unless label.nil? || label.empty?
+              included_work['author'] = author unless author.nil? || author.empty?
+              included_work['title'] = title unless title.nil? || title.empty?
+              included_work['title_nonfiling'] = title_nonfiling unless title_nonfiling.nil? || title_nonfiling.empty?
+              included_work['title_variation'] = title_variation unless title_variation.nil? || title_variation.empty?
+              included_work['details'] = details unless details.nil? || details.empty?
+              included_work['issn'] = issn unless issn.nil? || issn.empty?
+              included_work['isbn'] = isbn unless isbn.nil? || isbn.empty?
+              included_work['other_ids'] = other_ids unless other_ids.nil? || other_ids.empty?
+              included_work['display'] = display unless display.nil? || display.empty?
+
+              acc << included_work unless included_work.empty?
+            end
+          end
+        end
+
+        ################################################
+        # Type: Included Work
+        ######
+        def included_work_type(field)
+          case field.tag
+          when /(700|710|711)/
+            return unless passes_constraint_has_tk?(field)
+            'included'
+          when /(730|740)/
+            'included'
+          when '774'
+            return unless passes_constraint_has_ts?(field)
+            'included'
+          end
+        end
+
+        ################################################
+        # Label: Included Work
+        ######
+        def included_work_label(field)
+          label = []
+          case field.tag
+          when /(700|710|711)/
+            return unless passes_constraint_has_tk?(field)
+            label << collect_subfield_3_label_values(field)
+            label << process_subfield_i(field)
+          when '730'
+            label << collect_subfield_3_label_values(field)
+            label << process_subfield_i(field)
+          when '774'
+            return unless passes_constraint_has_ts?(field) && field.indicator1 == '0'
+            label << process_subfield_i(field)
+          end
+          label.flatten.compact.select { |l| !l.empty? }.join(": ")
+        end
+
+        def collect_subfield_3_label_values(field)
+          field.subfields.select { |sf| sf.code == '3' }.map(&:value).first.to_s.chomp(":")
+        end
+
+        def process_subfield_i(field)
+          field.subfields.select { |sf| sf.code == 'i' }
+               .map { |i| i.value.gsub(/\s\((work|expression|manifestation|item)\)/, '')
+                                 .chomp(":")
+                                 .capitalize
+                                 .gsub('Container of', 'Contains')
+                                 .gsub('Contained in', "In") }
+        end
+
+        ################################################
+        # Author: Included Work
```

Comment:
This same code should be reusable for the other *_work fields (except for the 774 field piece --- but that logic should also be extendable to the other 76X-78X field processing)